client/core/core_test.go

```diff
@@ -785,7 +785,7 @@ func newTestRig() *testRig {
 			lockTimeTaker: dex.LockTimeTaker(dex.Testnet),
 			lockTimeMaker: dex.LockTimeMaker(dex.Testnet),
 			wallets:       make(map[uint32]*xcWallet),
-			blockWaiters:  make(map[uint64]*blockWaiter),
+			blockWaiters:  make(map[string]*blockWaiter),
```

Comment:
Since `blockWaiters` is only used for registration fee txns, I believe `account.AccountID` would be a more appropriate key for the map given you can't register twice with the same account ID, at present anyway.  Or even the DEX's pubkey, which would be a `[secp256k1.PubKeyBytesLenCompressed]byte`.
But it's OK to leave as is.

Reply:
What would you think about using dex host as key?

Reply:
Maybe.  I kinda think that some broader refactoring is warranted for future work though.  `blockWaiters` makes it seem like a generic waiter, and it is, but really it's just used for watching fee txns for confirms.  So if this is made explicitly into a `regFeeWaiters` map then it would make more sense to key it on the DEX server's pubkey or host.

I think this work fixes issue https://github.com/decred/dcrdex/issues/552 so it can stop there for now.

Reply:
But the reason I suggested the pubkey is because in general (1) strings make less efficient map keys than say arrays or structs even though any comparable type can be used, and (2) the semantics of the string as a key are not obvious at a glance while when you see `map[account.AccountID]whatever` it's perfectly clear.